backend/chromeApiServer/src/main/java/com/canvas/service/EvaluationService.java

```diff
@@ -114,18 +114,68 @@ public ResponseEntity<CommandOutput> executeCodeFile(ExtensionUser user) throws
         }
     }
 
+    /**
```

Comment:
That makes sense. I guess my only question is where should creating the makefile (`writeMakefile` method) lie? Eval service and a new service (such as SubmissionDirectoryService) both need to pull the makefile from canvas. Maybe a static method in one of the two classes?

Reply:
Anything to do with extracting files from canvas should fall on the `CanvasClientService`, since that's interacting with Canvas. You can call `CanvasClientService` in the new class for `generateSubmissionDirectory` to get the makefile from canvas and store it as an object (which I believe you created a model for in `SubmissionFile.java` and `Submission.java`) or as is in the directory being created with the new code. That file doesn't need to be executed until the program is being run right? So, writing the file should only happen with the `executeCodeFile` method is called, which may need to be updated to reference the directory created by `generateSubmissionDirectory`. With that said, a few classes and method would need to be updated to pass this information around. Let me know if that doesn't make sense and we can hop on a call.

Reply:
finished refactor to `SubmissionDirectoryService`